### ISSUE TITLE
Encrypt the folder after backup, also decrypt it during restore 

### ISSUE THREAD

**Issue by rajib-raiyat** [laurent22/rsync-time-backup#256]
Is there any encryption and decryption support available?

**Comment by psmanek** [comment]
No. You can encrypt whole drive, for example using FileVault on mac.

**Comment by rajib-raiyat** [comment]
**Isn't it a good feature** that we include encryption and decryption?

**Comment by rajib-raiyat** [comment]
Most people like you don't encrypt the entire drive, but if they want to backup their data to a partition on another hard drive (not SSH), it can be very useful. Also in this script we can offer both options (backup with encryption or backup without encryption) to the user. It could be a very good feature to give an additional security option from our (this script) end.

**Comment by psmanek** [comment]
Backup solution should as easy as possible. If you make encrypted backup and after few years this script will no longer be developed, you may lose access to your backup files (because they are encrypted). Simplicity is the key.

Try BorgBackup or Restic if you want encrypted and deduplicated backups.

**Comment by rajib-raiyat** [comment]
I agree with your opinion that backup solutions should be simple and easy to use. However, it's also true that if the backup script is no longer supported after a few years, users may not be able to access their backup files. One potential solution is to encrypt the backup files with a password and use a consistent encryption and decryption algorithm to ensure long-term accessibility and security.
